torch/csrc/dynamo/eval_frame.c

```diff
@@ -299,6 +305,20 @@ typedef struct cache_entry {
   struct cache_entry* next;
 } CacheEntry;
 
+// ExtraState encasulates CacheEntry and FrameState. ExtraState is the highest
+// level of abstraction of what is stored on the extra code object. Previously,
+// we saved different parts on different extra indexes.  We prefer this way
+// because of cleaner abstraction and faster SetExtra access.
+typedef struct {
+  // Cache entry for the code object
+  CacheEntry* cache_entry;
+  // Frame state to detect dynamic shape dims
+  FrameState* frame_state;
+} ExtraState;
```

Comment:
I wonder if you do want to turn this into an honest to goodness PyObject eventually, as you did in your other PR.

Reply:
I am not planning to. It was needed because I was using PyDict earlier and i needed PyObject equivalent of CacheEntry then. Atleast for this PR, I dont need it.